src/sign/signature.rs

```diff
@@ -1371,6 +1371,79 @@ mod test {
         }
     }
 
+    #[should_panic]
+    #[test]
+    fn signer_index_should_match_public_key() {
+        let params = ThresholdParameters::<Secp256k1Sha256>::new(2, 1);
+        let message = b"This is a test of the tsunami alert system. This is only a test.";
+
+        let p1_sk = IndividualSigningKey {
+            index: 1,
+            key: Fr::rand(&mut OsRng),
+        };
+
+        let (p1_public_comshares, _) =
+            generate_commitment_share_lists::<Secp256k1Sha256>(&mut OsRng, &p1_sk, 1).unwrap();
+
+        let mut aggregator = SignatureAggregator::new(
+            params,
+            GroupVerifyingKey::new(Projective::zero()),
+            &message[..],
+        );
+
+        // The aggregator should panic as the participant index does not match the public key index.
```

Comment:
A general comment on the library: is it really fair that a library crashes the whole process like this? I think that should be a decision for the app that uses the library. I guess there are catastrophic situations where the library *must* crash, but unless this is one of them, I think it would be better to return error.